lib/puppet-lint/bin.rb

```diff
@@ -57,14 +71,15 @@ def run
 
       return_val = 0
       ignore_paths = PuppetLint.configuration.ignore_paths
+      all_problems = []
 
       puts '[' if PuppetLint.configuration.json
```

Comment:
If you're already collecting all problems in an array, you can remove this part here and properly serialize JSON at the end.

Reply:
yes, fixed and it is better because it is now pretty print,
before change manual adding [ and , will not align well.